server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/node/StarTree.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.index.compositeindex.datacube.startree.node;
+
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.RandomAccessInput;
+import org.opensearch.index.compositeindex.datacube.startree.fileformats.meta.StarTreeMetadata;
+import org.opensearch.index.compositeindex.datacube.startree.fileformats.node.FixedLengthStarTreeNode;
+
+import java.io.IOException;
+
+/**
+ * Off heap implementation of the star-tree.
+ *
+ * @opensearch.experimental
+ */
+public class StarTree {
```

Comment:
Do we have a corresponding on-heap implementation. I don't see any interface though

Reply:
IMO, Node should be on-heap/off-heap aware and we may not need a separate StarTree class at all (Just a static method within StarTreeNode should achieve this).

Reply:
StarTree class takes an IndexInput slice instead of a random access input.
StarTree gives proper structure, and allows the scope of a different implementation of Nodes to exists incase it changes in the future. Currently we have FixedLengthNodes, that can change in the future. But based on the version of the star tree metadata, we can toggle around the actual implementation without changing the interface.

With multiple star trees, we slice the index input, and initialize the corresponding root node.

Reply:
Implemented an StarTreeFactory instead of StarTree interface